torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,405 @@
+    12. Fruits Vegetables Flowers
+    13. Potatoes
+    14. Leguminous Fodder
+    15. Soybeans
+    16. Orchard
+    17. Mixed Cereal
+    18. Sorghum
+    19. Void Label
+
+    If you use this dataset in your research, please cite the following papers:
+
+    * https://doi.org/10.1109/ICCV48922.2021.00483
+    * https://doi.org/10.1016/j.isprsjprs.2022.03.012
+
+    .. versionadded:: 0.5
+    """
+
+    classes = [
+        "background",  # all non-agricultural land
+        "meadow",
+        "soft_winter_wheat",
+        "corn",
+        "winter_barley",
+        "winter_rapeseed",
+        "spring_barley",
+        "sunflower",
+        "grapevine",
+        "beet",
+        "winter_triticale",
+        "winter_durum_wheat",
+        "fruits_vegetables_flowers",
+        "potatoes",
+        "leguminous_fodder",
+        "soybeans",
+        "orchard",
+        "mixed_cereal",
+        "sorghum",
+        "void_label",  # for parcels mostly outside their patch
+    ]
+    cmap = {
+        0: (0, 0, 0, 255),
+        1: (174, 199, 232, 255),
+        2: (255, 127, 14, 255),
+        3: (255, 187, 120, 255),
+        4: (44, 160, 44, 255),
+        5: (152, 223, 138, 255),
+        6: (214, 39, 40, 255),
+        7: (255, 152, 150, 255),
+        8: (148, 103, 189, 255),
+        9: (197, 176, 213, 255),
+        10: (140, 86, 75, 255),
+        11: (196, 156, 148, 255),
+        12: (227, 119, 194, 255),
+        13: (247, 182, 210, 255),
+        14: (127, 127, 127, 255),
+        15: (199, 199, 199, 255),
+        16: (188, 189, 34, 255),
+        17: (219, 219, 141, 255),
+        18: (23, 190, 207, 255),
+        19: (255, 255, 255, 255),
+    }
+    directory = "PASTIS-R"
+    filename = "PASTIS-R.zip"
+    url = "https://zenodo.org/record/5735646/files/PASTIS-R.zip?download=1"
+    md5 = "4887513d6c2d2b07fa935d325bd53e09"
+    prefix = {
+        "s2": os.path.join("DATA_S2", "S2_"),
+        "s1a": os.path.join("DATA_S1A", "S1A_"),
+        "s1d": os.path.join("DATA_S1D", "S1D_"),
+        "semantic": os.path.join("ANNOTATIONS", "TARGET_"),
+        "instance": os.path.join("INSTANCE_ANNOTATIONS", "INSTANCES_"),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        folds: Sequence[int] = (0, 1, 2, 3, 4),
+        bands: str = "s2",
+        mode: str = "semantic",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new PASTIS dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            folds: a sequence of integers from 0 to 4 specifying which of the five
+                dataset folds to include
+            bands: load Sentinel-1 ascending path data (s1a), Sentinel-1 descending path
+                data (s1d), or Sentinel-2 data (s2)
+            mode: load semantic (semantic) or instance (instance) annotations
```

Comment:
Would "task" be a better parameter name? Also might change this to:
```suggestion
            mode: load "semantic" or "instance" segmentation annotations
```